Release Notes for August 2024 (#3250)
MicrosoftDocs/edge-developer: microsoft-edge/webview2/release-notes/index.md

###### SDK-only
-->

<!-- end of Sep 2024 Release SDK -->
<!-- end of Sep 2024 Prerelease SDK -->
<!-- end of template -->

<!-- ====================================================================== -->
## 1.0.2739.15

Release Date: August 26, 2024

[NuGet package for WebView2 SDK 1.0.2739.15](https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.2739.15)

For full API compatibility, this Release version of the WebView2 SDK requires WebView2 Runtime version 128.0.2739.15 or higher.

<!-- ------------------------------ -->
#### Promotions

The following APIs have been promoted to Stable and are now included in this Release SDK.

<!-- ------------------------------ -->

> * Added `Notification` APIs that support web notifications, including non-persistent notifications and persistent notifications.  A _non-persistent notification_ is a notification without an associated service worker registration.  A _persistent notification_ is a notification with an associated service worker registration.  You should be able to handle<!-- todo: "Your app should handle"?  "Your app can optionally handle"? --> notification permission requests, and further listen to<!-- todo: "and then listen for"? --> `Notification Received` events to optionally handle the notifications themselves.<!-- todo: delete "themselves"?  does "should" contradict "optionally"? -->

Reviewer: The notification APIs only cover the non-persistent notifications but not persistent notifications. The summary should be the same/similar to the description in 2730-prerelease.

Author: result:
```
* Added support for Web Notification, for non-persistent notifications.  The `NotificationReceived` event for `CoreWebView2` controls web notification handling, allowing customization or suppression by the host app.  Unhandled notifications default to WebView2's UI.
```

<!-- Methods -->
   * [CoreWebView2Notification.ReportClicked Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2notification?view=webview2-winrt-1.0.2739.15&preserve-view=true#reportclicked)
   * [CoreWebView2Notification.ReportClosed Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2notification?view=webview2-winrt-1.0.2739.15&preserve-view=true#reportclosed)
   * [CoreWebView2Notification.ReportShown Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2notification?view=webview2-winrt-1.0.2739.15&preserve-view=true#reportshown)
<!-- Events -->
   * [CoreWebView2Notification.CloseRequested Event](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2notification?view=webview2-winrt-1.0.2739.15&preserve-view=true#closerequested)

* [CoreWebView2NotificationReceivedEventArgs Class](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2notificationreceivedeventargs?view=webview2-winrt-1.0.2739.15&preserve-view=true)

>    * [CoreWebView2NotificationReceivedEventArgs.Cancel Property](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2notificationreceivedeventargs?view=webview2-winrt-1.0.2739.15&preserve-view=true#cancel)

Author: should the 3 tabs have `CoreWebView2NotificationReceivedEventArgs.Cancel`, or not?  
@peiche-jessica @nishitha-burman -  The research below indicates that I should delete the 1 Cancel that came in, here.  it came in in the orig [commit](https://github.com/MicrosoftDocs/edge-developer/pull/3250/commits/7129e6e96f36d1503f41b2b086f371d5ffbe5588)

it's currently inconsistent:
.NET tab: lacks Cancel.
WinRT tab: has Cancel.
Win32 tab: lacks Cancel

.NET Ref: doesn't exist yet; instead, check live latest (promoted to Stable in Prerelease): 
https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.core.corewebview2notificationreceivedeventargs?view=webview2-dotnet-1.0.2730-prerelease - lacks Cancel.
WinRT Ref: check open PR:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2notificationreceivedeventargs?view=webview2-winrt-preview&branch=pr-en-us-64 - lacks Cancel.
Win32 Ref: check open PR:
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2notificationreceivedeventargs?view=webview2-win32-preview&branch=pr-en-us-97 - lacks Cancel

we're in the section "promoted to Stable in Release", therefore earlier RelNotes section exists, and earlier Ref exists: 
* promoted to Stable in Prerelease.
* Experimental in Prerelease.

rendered article section: 2nd tab set:
https://learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3250#promotions-1

there are no other mentions of
CoreWebView2NotificationReceivedEventArgs.Cancel
in relnotes file.  only this incoming mention.

Reviewer: Please remove accordingly. Thanks.

Author: removed CoreWebView2NotificationReceivedEventArgs.Cancel

   * [ICoreWebView2NotificationReceivedEventArgs::GetDeferral](/microsoft-edge/webview2/reference/win32/icorewebview2notificationreceivedeventargs?view=webview2-1.0.2739.15&preserve-view=true#getdeferral)
   * [ICoreWebView2NotificationReceivedEventArgs::put_Handled](/microsoft-edge/webview2/reference/win32/icorewebview2notificationreceivedeventargs?view=webview2-1.0.2739.15&preserve-view=true#put_handled)

* [ICoreWebView2NotificationReceivedEventHandler](/microsoft-edge/webview2/reference/win32/icorewebview2notificationreceivedeventhandler?view=webview2-1.0.2739.15&preserve-view=true)<!-- Win32-only -->

* [`COREWEBVIEW2_TEXT_DIRECTION_KIND` Enum](/microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.2739.15&preserve-view=true#corewebview2_text_direction_kind)
   * `COREWEBVIEW2_TEXT_DIRECTION_KIND_DEFAULT`
   * `COREWEBVIEW2_TEXT_DIRECTION_KIND_LEFT_TO_RIGHT`
   * `COREWEBVIEW2_TEXT_DIRECTION_KIND_RIGHT_TO_LEFT`

---

<!-- ------------------------------ -->

> * Added the `SaveAs` APIs that<!-- todo: ", which"?  is this all SaveAs apis, or just a new group of them? --> allow you to programmatically perform the **Save as** operation.  You can use these APIs to block the default **Save as** dialog, and then either save silently, or build your own UI for **Save as**.  These APIs pertain only to the **Save as** dialog, not the **Download** dialog, which continues to use the existing Download APIs. 

Author: check older wording in:
* promoted to Stable in Prerelease:

Added `SaveAs` APIs that allow you to programmatically perform the **Save as** operation.  You can use these APIs to block the default **Save as** dialog, and then either save silently, or build your own UI for **Save as**.  These APIs pertain only to the **Save as** dialog, not the **Download** dialog, which continues to use the existing Download APIs.

* Experimental in Prerelease:

Added `SaveAs` APIs that allow you to programmatically perform the **Save as** operation.  You can use these APIs to block the default **Save as** dialog, and then either save silently, or build your own UI for **Save as**.  These APIs pertain only to the **Save as** dialog, not the **Download** dialog, which continues to use the existing Download APIs.

Author: result: used exact copy of an older desc, instead of this PR's orig commit that adds "the" and "adn".

* [ICoreWebView2ShowSaveAsUICompletedHandler](/microsoft-edge/webview2/reference/win32/icorewebview2showsaveasuicompletedhandler?view=webview2-1.0.2739.15&preserve-view=true)<!-- Win32-only -->

* [`COREWEBVIEW2_SAVE_AS_KIND` Enum](/microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.2739.15&preserve-view=true#corewebview2_file_save_as_kind)
   * `COREWEBVIEW2_SAVE_AS_KIND_DEFAULT`
   * `COREWEBVIEW2_SAVE_AS_KIND_HTML_ONLY`
   * `COREWEBVIEW2_SAVE_AS_KIND_SINGLE_FILE`
   * `COREWEBVIEW2_SAVE_AS_KIND_COMPLETE`

* [`COREWEBVIEW2_SAVE_AS_UI_RESULT` Enum](/microsoft-edge/webview2/reference/win32/webview2-idl?view=webview2-1.0.2739.15&preserve-view=true#corewebview2_save_as_ui_result_kind)
   * `COREWEBVIEW2_SAVE_AS_UI_RESULT_SUCCESS`
   * `COREWEBVIEW2_SAVE_AS_UI_RESULT_INVALID_PATH`
   * `COREWEBVIEW2_SAVE_AS_UI_RESULT_FILE_ALREADY_EXISTS`
   * `COREWEBVIEW2_SAVE_AS_UI_RESULT_KIND_NOT_SUPPORTED`
   * `COREWEBVIEW2_SAVE_AS_UI_RESULT_CANCELLED`
   
---

<!-- ------------------------------ -->
#### Bug fixes

> <!-- todo: populate, or note None, or link to other section (expand the latter's heading to specify version #) -->

Author: Which to do?
* Populate this section.
* State that there are no bug fixes for this Release SDK.
* Link to the Prerelease Bug Fixes section.

@peiche-jessica @nishitha-burman 

rendered article section (Release, empty):
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3250&tabs=dotnetcsharp#bug-fixes

rendered article section (Prerelease):
https://review.learn.microsoft.com/en-us/microsoft-edge/webview2/release-notes/?branch=pr-en-us-3250&tabs=dotnetcsharp#bug-fixes-1

Reviewer 2: State that there are no bug fixes for this Release SDK.

<!-- ---------- -->
###### Runtime and SDK

<!-- ---------- -->
###### Runtime-only

<!-- ---------- -->
###### SDK-only

<!-- end of Aug 2024 Release SDK -->

<!-- ====================================================================== -->
## 1.0.2783-prerelease

Release Date: August 26, 2024

[NuGet package for WebView2 SDK 1.0.2783-prerelease](https://www.nuget.org/packages/Microsoft.Web.WebView2/1.0.2783-prerelease)

For full API compatibility, this Prerelease version of the WebView2 SDK requires the WebView2 Runtime that ships with Microsoft Edge version 129.0.2783.0 or higher.

<!-- ------------------------------ -->
#### Experimental APIs

No Experimental APIs have been added in this Prerelease SDK.

---

<!-- ------------------------------ -->
#### Promotions

No APIs have been promoted from Experimental to Stable in this Prerelease SDK.

---

<!-- ------------------------------ -->
#### Bug fixes

<!-- ---------- -->
###### Runtime and SDK

> * Re-enabled the default behavior of `SetUserAgent`: by default, `SetUserAgent` is effective for cross-origin iframes. 

Author: wording per orig commit: 
Re-enables the default behavior of SetUserAgent being effective for cross-origin iframes.

clarify "being effective for" (ambiguous) -- is the revised line a clear, unambiguous, correct wording? --
Re-enabled the default behavior of `SetUserAgent`: by default, `SetUserAgent` is effective for cross-origin iframes. 

@peiche-jessica @nishitha-burman 

rendered:

Reviewer 2: Sounds good to me.

<!-- todo: clarify "being effective for" -->

<!-- ---------- -->
###### Runtime-only

* Enabled the interactive dragging feature by default.  See `edge-webview-interactive-dragging` in [WebView2 browser flags](../concepts/webview-features-flags.md).
* Disabled `IsolateSandboxedIframes` for WebView2.
* Fixed an issue where WebView creation fails when multiple instances are launched at the same time.  ([Issue #4731](https://github.com/MicrosoftEdge/WebView2Feedback/issues/4731))

> * Fixed a bug in WinRT JavaScript projection where caching existing properties in Proxy and Function like name<!-- todo: clarify/format, give example api names.  does name literally contain "Proxy" or "Function"?  or a name that's like the name of the proxy or the function? --> caused an error due to name collision.

Author: ```suggestion
* Fixed a bug in WinRT JavaScript projection where caching existing properties in objects whose name contains `Proxy` or `Function` caused an error due to name collision.
```
clarify.  does name literally contain "Proxy" or "Function"?  or a name that's like the name of the proxy or the function?

orig commit's wording:
* Fixed a bug in WinRT JavaScript projection where caching existing properties in Proxy and Function like name caused an error due to name collision.
this Suggestion's wording:
* Fixed a bug in WinRT JavaScript projection where caching existing properties in objects whose name contains `Proxy` or `Function` caused an error due to name collision.
alt wording:
* Fixed a bug in WinRT JavaScript projection where caching existing properties in objects whose name is like the name of the proxy or the function caused an error due to name collision.

@peiche-jessica @nishitha-burman

Reviewer 2: Suggestion sounds good to me, thanks.

* Fixed a bug where the WebView2 control became the wrong size after disconnecting and reconnecting a monitor.
* Fixed an issue where "mailto:" links leave an untitled popup window open, instead of automatically closing the popup window.

<!-- ---------- -->
###### SDK-only

* C# WinRT projection now works on UWP.

> * Fix for `GeneratedFilesDir` showing up in Visual Studio for C# WinRT projection.<!-- todo: clarify: "`GeneratedFilesDir` no longer appears in Visual Studio during C# WinRT projection"? -->

Author: ```suggestion
* Fixed `GeneratedFilesDir` so that it no longer appears in Visual Studio for C# WinRT projection.
```
clarify.  tech review this proposed rewording @peiche-jessica @nishitha-burman 
* avoid "showing up", use standard technical wording.
* what sort of thing is "GeneratedFilesDir" - an API item that's listed as a member of list of projected items, shown as a line of code?
* avoid "Fix for", prefer "Fixed"

Reviewer 2: How about: Fixed an issue that ensures `GeneratedFileDir` no longer appears in Visual Studio for C# WinRT projection.

Author: result:
Fixed an issue to ensure that `GeneratedFilesDir` no longer appears in Visual Studio for C# WinRT projection.
